src/System.Private.Uri/tests/FunctionalTests/IdnDnsSafeHostTest.cs

```diff
@@ -18,6 +18,7 @@ public void IdnDnsSafeHost_IdnOffWithBuiltInScheme_Success()
 
             string dns = test.DnsSafeHost;
 
+            // string.Contains(char) is .NetCore2.1+ specific
```

Comment:
Comments seem even less useful for tests where we don't care about perf to this level anyway.

Reply:
See previous response. Do you want me to revert the comments in tests/asserts?

Reply:
Reverted all comments in test/assert. Let me know if you want me to revert all comments

Reply:
@ahsonkhan thoughts? The only reason to not revert them all is if they have `using System.Linq;` then I believe that Contains would compile to a slow Linq version.

Reply:
> then I believe that Contains would compile to a slow Linq version.

Imo, that's a good enough reason to keep the comment.

That Linq API is quite slow, especially due to boxing:

|         Method | IsDataCompact |   TestCase |      Mean |     Error |    StdDev |  Gen 0 | Allocated |
|--------------- |-------------- |----------- |----------:|----------:|----------:|-------:|----------:|
| StringContains |          True | HelloWorld |  2.453 ns | 0.3580 ns | 0.0930 ns |      - |       0 B |
|   LinqContains |          True | HelloWorld | 31.012 ns | 2.0902 ns | 0.5429 ns | 0.0076 |      32 B |

Let's make sure we aren't accidentally calling the Linq.Contains API in hot paths.

Reply:
In my cursory look, I was only able to find two possible candidates where we could use IndexOf instead (and avoid using [Linq.Contains](https://apisof.net/catalog/System.Linq.Enumerable.Contains%3CTSource%3E(IEnumerable%3CTSource%3E,TSource)):
https://github.com/dotnet/corefx/blob/a99a57f59f8152099e7da3e2c623568ee9cc016e/src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs#L587
https://github.com/dotnet/corefx/blob/a99a57f59f8152099e7da3e2c623568ee9cc016e/src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs#L783

The ones in System.Net.WebHeaderCollection are fine since we target netcoreapp for that project:
https://github.com/dotnet/corefx/blob/a99a57f59f8152099e7da3e2c623568ee9cc016e/src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs#L24
https://github.com/dotnet/corefx/blob/a99a57f59f8152099e7da3e2c623568ee9cc016e/src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs#L68
https://github.com/dotnet/corefx/blob/a99a57f59f8152099e7da3e2c623568ee9cc016e/src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs#L71

Reply:
There is another perf comparison here https://github.com/dotnet/corefx/issues/25094 where I originally added the overload. I saw 20x

Reply:
> 20x
Wow. I still have a task on my backlog to see what the per improvement is with this change. I will get to it after the units.

Reply:
> two possible candidates

OK, I will try update those too. 
BTW, I hadn't though to look for existing callsites using the pattern `.Contains("<single char>")`. Your 'SNIProxy` note made me realize that and so I have found a few more places I can change.